app/services/csv_imports/gias_import.rb

```diff
@@ -0,0 +1,44 @@
+require "csv"
+
+module CsvImports
+  class GiasImport
+    include ServicePattern
+
+    attr_reader :csv_path, :logger
+
+    def initialize(csv_path)
+      @csv_path = csv_path
+      @logger = Logger.new($stdout)
+    end
+
+    def call
+      upserted = 0
+      rows = 0
+      errors = []
+
+      CSV
+        .foreach(csv_path, headers: true, encoding: "iso-8859-1:utf-8")
+        .with_index(2) do |school, row_number|
+          next if school["CloseDate"].present?
+
+          begin
+            GiasSchool.find_or_initialize_by(urn: school["URN"]).update!(
```

Comment:
I leave it to you but you may want to do a bulk update here and not loop through item.

Reply:
Will this take four minutes every time it is run (and we'll be running it daily?) If so, I think there is an argument to be made for running it as a series of smaller jobs, maybe batching 1000 rows at a time. But perhaps that is beyond the scope of this ticket / irrelevant given the relatively small amount of traffic we anticipate on the site.

Reply:
I've gone with a bulk update. On my machine importing almost 24k schools took about 20 seconds, so we should be fine now 🤞🏻